matsim/src/test/java/org/matsim/core/router/NetworkRoutingInclAccessEgressModuleTest.java

```diff
@@ -49,86 +55,101 @@ private static Scenario createScenario() {
+        network.addLink(longButFast2);
+
+        Link endLInk = createLink(END_LINK, n4, n5, 50, FAST_SPEED, modes, factory);
         network.addLink(endLInk);
+
         return scenario;
     }
 
-    private static Person createPerson(String id, Coord homeCoord, Coord workCoord, String mode, PopulationFactory factory) {
+    private static Link createLink(String id, Node from, Node to, double length, double freespeed, Set<String> modes, NetworkFactory factory) {
+
+        Link link = factory.createLink(Id.createLinkId(id), from, to);
+        link.setLength(length);
+        link.setFreespeed(freespeed);
+        link.setAllowedModes(modes);
+        link.setCapacity(100000); // get capacity constraints out of the way
+        return link;
+    }
+
+    private static Person createPerson(String id, String mode, PopulationFactory factory) {
 
         Person person = factory.createPerson(Id.createPersonId(id));
 
         Plan plan = factory.createPlan();
-        Activity home1 = factory.createActivityFromCoord("home", homeCoord);
+        Activity home1 = factory.createActivityFromCoord("home", HOME);
         home1.setEndTime(0);
-        home1.setLinkId(Id.createLinkId("start"));
+        home1.setLinkId(Id.createLinkId(START_LINK));
         plan.addActivity(home1);
 
         Leg toWork = factory.createLeg(mode);
         plan.addLeg(toWork);
 
-        Activity work = factory.createActivityFromCoord("work", workCoord);
+        Activity work = factory.createActivityFromCoord("work", WORK);
         work.setEndTime(300);
-        work.setLinkId(Id.createLinkId("end"));
+        work.setLinkId(Id.createLinkId(END_LINK));
         plan.addActivity(work);
 
         person.addPlan(plan);
 
         return person;
     }
 
-    private static VehicleType createVehicleType(String name, double maxVelocity, int capacity, VehiclesFactory factory) {
+    private static VehicleType createVehicleType(String name, double maxVelocity, VehiclesFactory factory) {
 
         VehicleType result = factory.createVehicleType(Id.create(name, VehicleType.class));
         VehicleCapacity vehicleCapacity = new VehicleCapacityImpl();
-        vehicleCapacity.setSeats(capacity);
+        vehicleCapacity.setSeats(1);
         result.setCapacity(vehicleCapacity);
         result.setMaximumVelocity(maxVelocity);
         return result;
     }
 
+    private static List<Id<Link>> getNetworkRoute(List<PlanElement> elements) {
+        return elements.stream()
+                .filter(element -> element instanceof Leg)
+                .map(element -> (Leg) element)
+                .filter(leg -> !leg.getMode().equals(TransportMode.non_network_walk))
+                .flatMap(leg -> ((NetworkRoute) leg.getRoute()).getLinkIds().stream())
+                .collect(Collectors.toList());
+    }
+
     @Test
-    public void calcRoute() {
+    public void calcRoute_speedOfIndividualVehicle() {
```

Comment:
I've put an integration test into this class to test, whether the router respects the vehicles and their speeds. Is this the right place @kainagel ?

Reply:
I guess that for tests we don't have a lot of convention.  I move tests if by moving it I can make some methods package-private, but that is about it.